Movecraft/src/main/java/net/countercraft/movecraft/async/rotation/RotationTask.java

```diff
@@ -195,150 +194,145 @@ protected void execute() {
-            for(Entity entity : craft.getWorld().getNearbyEntities(midpoint,
-                    oldHitBox.getXLength() / 2.0 + 1,
-                    oldHitBox.getYLength() / 2.0 + 2,
-                    oldHitBox.getZLength() / 2.0 + 1)) {
-
-                if (entity instanceof HumanEntity) {
-                    InventoryView inventoryView = ((HumanEntity) entity).getOpenInventory();
-                    if (inventoryView.getType() != InventoryType.CRAFTING) {
-                        Location l = Movecraft.getInstance().getWorldHandler().getAccessLocation(inventoryView);
-                        if (l != null) {
-                            MovecraftLocation location = new MovecraftLocation(l.getBlockX(), l.getBlockY(), l.getBlockZ());
-                            if (oldHitBox.contains(location)) {
-                                location = MathUtils.rotateVec(rotation, location.subtract(originPoint)).add(originPoint);
-                                updates.add(new AccessLocationUpdateCommand(inventoryView, location.toBukkit(w)));
-                            }
-                        }
-                    }
-                }
+        rotateEntitiesOnCraft(tOP);
 
-                if (!craft.getType().getBoolProperty(CraftType.ONLY_MOVE_PLAYERS) || (
-                        (entity.getType() == EntityType.PLAYER || entity.getType() == EntityType.PRIMED_TNT)
-                                && !(craft instanceof SinkingCraft)
-                )) {
-                    // Player is onboard this craft
-
-                    Location adjustedPLoc = entity.getLocation().subtract(tOP);
-
-                    double[] rotatedCoords = MathUtils.rotateVecNoRound(rotation,
-                            adjustedPLoc.getX(), adjustedPLoc.getZ());
-                    float newYaw = rotation == MovecraftRotation.CLOCKWISE ? 90F : -90F;
-
-                    CraftTeleportEntityEvent e = new CraftTeleportEntityEvent(craft, entity);
-                    Bukkit.getServer().getPluginManager().callEvent(e);
-                    if (e.isCancelled())
-                        continue;
-
-                    EntityUpdateCommand eUp = new EntityUpdateCommand(entity,
-                            rotatedCoords[0] + tOP.getX() - entity.getLocation().getX(),
-                            0,
-                            rotatedCoords[1] + tOP.getZ() - entity.getLocation().getZ(),
-                            newYaw,
-                            0
-                    );
-                    updates.add(eUp);
-                }
+        Craft craft1 = getCraft();
+        if (craft1.getCruising()) {
+            CruiseDirection direction = craft1.getCruiseDirection();
+            switch (rotation) {
+                case ANTICLOCKWISE -> craft1.setCruiseDirection(direction.getAnticlockwiseRotation());
+                case CLOCKWISE -> craft1.setCruiseDirection(direction.getClockwiseRotation());
```

Comment:
See the below comment, this can be simplified.

Reply:
Considered that but for some reason my brain decided not to add it, will add it soon